man/systemd-cryptenroll.xml

```diff
@@ -387,6 +387,28 @@
         enrollment to.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--tpm2-external-key=</option><replaceable>PATH</replaceable></term>
+
+        <listitem><para>Enroll an external TPM2 security chip. Expects a path referring to the TPM2 public
+        key in TPM2B_PUBLIC format. This cannot be used with <option>--tpm2-device</option>, and this does
```

Comment:
why in TPM2B_PUBLIC format? we so far are always using PEM format when interfacing with the user. Why use TPM2B_PUBLIC here? why not convert on the fly? all these tpm concepts are hard to grok anyway, i'd rather not confuse users further by pushing two different key marshallings on them.

please suffix `--tpm2-device=` with `=`

Reply:
> why in TPM2B_PUBLIC format? we so far are always using PEM format when interfacing with the user. Why use TPM2B_PUBLIC here? why not convert on the fly? all these tpm concepts are hard to grok anyway, i'd rather not confuse users further by pushing two different key marshallings on them.

we do need more than just the RSA/ECC key data for the calculations, we also need the name hash alg (e.g. SHA256) as well as the symmetric alg (e.g. AES-128-CFB). And finally, since we now store the serialized SRK, we must have the entire thing, since that (plus its "name" and handle index) is what goes into the serialized object.

> 
> please suffix `--tpm2-device=` with `=`

ack, updated